Add terms to glossary (#290)
microsoft/vcpkg-docs: vcpkg/concepts/glossary.md

discussed by the community. Because of that, and to avoid confusion and
ambiguity this article provides specific meanings for these terms.

## A

### Artifact

Files created during the build process, essentially the output of your build
process. Artifacts include libraries, executables, headers, logs, and other

> files that are generated by compiling the source code.

Reviewer: > Download artifact

I can smell the confusion with regard to assets.

> files that are generated by compiling the source code.

Author: I changed the term to build artifact.

### Asset

An artifact that needs to be downloaded typically from the Internet and made
available locally, often the input of a build process. Assets include source
code archives, pre-built binaries, or other files that are required to build
a package.

## B

### Binary package

The installation output produced by vcpkg packaged into a restorable format.
Binary packages hold the build-output of a package (binaries, build system
integration files, usage documentation, license, and other files). Binary

> packages can be reused across different machines and platforms without rebuilding.

Author: I removed the sentence, since it's not entirely accurate.

## C

### Classic Mode

> A mode of vcpkg that installs packages into a centrally installed tree.

Reviewer: `Centrally/centralied` is a strong word, given that I can have any numbers of it. But it is not like `/usr`.
Maybe `shared` (in contrast to per-build manifest mode)?

is to install these packages [available during your projects
builds](../concepts/build-system-integration.md).

### Package manager

> A package manager is a tool that manages dependencies for a project used
> to set up installing, upgrading, using software libraries and tools. Package

Reviewer 2: ```suggestion
A package manager is a tool that manages dependencies for a project by
offering features to install, upgrade, and integrate software libraries and tools. Package
```

* Instructions to acquire, build if necessary, and install the package.

See the [ports documentation](../concepts/ports.md) to learn more.

## S

### System Package Manager

A package manager such as apt-get designed to be used with a particular
platform such as Linux, or brew for MacOS. System package managers install

> packages and libraries system wide, instead of for a particular project.

Reviewer 2: ```suggestion
packages and libraries system wide, rather than for a particular project.
```

## T

### Triplet

A set of properties or script applied to all libraries before vcpkg builds,

> essentially a profile. A triplet specifies the linkage type, build flags,

Reviewer 2: The problem with this definition is that it introduces a new term, what is a "profile"?
